src/main/java/no/fdk/utils/AssessmentUtils.java

```diff
@@ -214,7 +212,9 @@ private static Catalog extractCatalogFromModel(Model model, Resource datasetReso
         String catalogUri = null;
 
         if (datasetResource.hasProperty(DCTerms.publisher)) {
-            catalogId = extractPublisherIdFromPublisherResource(datasetResource.getPropertyResourceValue(DCTerms.publisher));
+            Resource catalogResource = datasetResource.getPropertyResourceValue(DCTerms.publisher);
+            catalogId = extractPublisherIdFromPublisherResource(catalogResource);
+            catalogUri = catalogResource != null ? catalogResource.getURI() : null;
```

Comment:
Kan være dette feiler om publisher er en blank node, sikkert greit å sikre med noe ish det her:
```suggestion
            catalogUri = catalogResource != null && catalogResource.isURIResource() ? catalogResource.getURI() : null;
```